src/server_manager/ui_components/outline-validated-input.html

```diff
@@ -0,0 +1,190 @@
+<!--
+  Copyright 2020 The Outline Authors
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<!-- outline-validated-input
+    This is an input, with a cancel and a save button, which performs client-side validation and has an
+    event-based hook for server-side evaluation.
+
+    Attributes:
+        * name: A unique name used to build element id's.  Default: null
+        * editable: Is this input editable?  Default: false
+        * visible: Is this input visible?  Default: false
+        * label: The label for the input.  Default: null
+        * value: The initial value entered in the input.  SHOULD be a valid value.  Default: null
+        * allowed-pattern: Regex describing what inputs are allowed.  Users will be prevented from
+            entering inputs which don't follow the pattern.  Default: ".*"
+        * max-length: The number of characters allowed in the input.  Default: Number.POSITIVE_INFINITY
+        * client-side-validator: A function which takes a string and returns either an empty string on success
+            or an error message on failure.  This function will be called on every keystroke. Default: () => ""
+        * event: The name of the event fired when the save button is tapped.  Passes the
+            current input value as "value" and the Polymer object as "ui".  The handler for this event
+            MUST call ui.setSavedState() or ui.setErrorState(message) depending on its result. Default: null
+ -->
+
+<link rel='import' href='./cloud-install-styles.html'>
+<link rel='import' href='./outline-server-settings-styles.html'>
+
+<dom-module id='outline-validated-input'>
+    <template>
+        <style include='cloud-install-styles'></style>
+        <style include='outline-server-settings-styles'></style>
+        <style>
+            [id$=Container] {
+                display: flex;
+                justify-content: space-between;
+                align-items: center;
+                max-width: 545px;
```

Comment:
nit: this seems arbitrary, better to use a percentage relative to the parent.

Reply:
This is simply preserving the current behavior.  Let's change that in a separate PR?